java/code/src/com/suse/manager/webui/utils/SaltRoster.java

```diff
@@ -68,9 +73,37 @@ public void addHost(String host, String user, Optional<String> passwd,
                 Arrays.asList(option)));
         timeout.ifPresent(value -> hostData.put("timeout", value));
         minionOpts.ifPresent(options -> hostData.put("minion_opts", options));
+        sshPreflightScriptPath.ifPresent(value -> hostData.put("ssh_pre_flight", value));
+        if (sshPreflightScriptPath.isPresent()) {
+            sshPreflightScriptArgs.ifPresent(value -> hostData.put("ssh_pre_flight_args",
+                Arrays.asList(value)));
```

Comment:
> hostData.put("ssh_pre_flight_args", Arrays.asList(value))

This is putting in the map a list with only one element, which is the args list (so `ssh_pre_flight_args` is a list of lists in the end). Is this what you wanted? Or should this just have been `hostData.put("ssh_pre_flight_args", value)`?

Reply:
Fixed. Thanks for pointing :+1: